### PR TITLE
[Hpriest] Making CoH baseline and adding Prayer Circle Talent

### PR DESCRIPTION
**Pull request by TJMoats**
a97MFS6

### CODE REVIEW

File: src/parser/priest/holy/modules/core/AbilityTracker.tsx
Comment (Dambroda) [inline src/parser/priest/holy/modules/core/AbilityTracker.tsx:3]: You can just use CoreAbilityTracker in your CombatLogParser if you aren't extending or changing any functionality

(this might be a lie but im 87% sure)

File: src/parser/shared/modules/resources/mana/ManaUsageGraph.js
Comment (Dambroda) [inline src/parser/shared/modules/resources/mana/ManaUsageGraph.js:90]: Good catch haha

File: src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx
Comment (Dambroda) [inline src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx:53]: I could be very wrong but in my head I was picturing that this would work:
```typescript
lastCastStartWasBuffed: boolean;

startPohCast(event: BeginCastEvent) {
    // entire comment revolves around this buff existing; I dont see it referenced anywhere in here so maybe it doesnt?
    this.lastCastStartWasBuffed = this.selectedCombatant.hasBuff(SPELLS.PRAYER_CIRCLE_BUFF.id); 
}

finishPohCast(event: CastEvent) {
  if (this.lastCastStartWasBuffed) {
    this.buffedCohCasts += 1;
  }
}
```

The logic being that the boolean value will always be the most "recent" begincast event for PoH, so you can use that when the cast finishes to see if the buff was active when the cast started

File: src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx
Comment (Dambroda) [inline src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx:39]: Not sure if you wanted to leave this in for testing; commenting in case it was accidental

**Review by Dambroda — COMMENTED** [review]
Some comments for thoughts, nothing here is blocking

File: src/parser/priest/holy/modules/talents/75/CircleOfHealing.tsx
Comment (TJMoats) [reply src/parser/priest/holy/modules/talents/75/CircleOfHealing.tsx:13]: I've re-added it as a general tracker.

File: src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx
Comment (TJMoats) [reply src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx]: Removed, thanks for catching that!

File: src/parser/priest/holy/modules/core/AbilityTracker.tsx
Comment (TJMoats) [reply src/parser/priest/holy/modules/core/AbilityTracker.tsx:3]: I'm planning on adding some stuff to the ability tracker, so I figured I'd just add the base module in this PR. I can remove it if you think I should for now.

File: src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx
Comment (TJMoats) [reply src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx:53]: I don't know the ID of the buff, so I figured that this was good enough for now.

Comment (Dambroda) [reply src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx:53]: Found a log with the talent:

https://wowanalyzer.com/report/y3pDAt2R1HLjdBNC/7-Heroic+Kael'thas+-+Wipe+7+(3:36)/Myixi/standard/events

321379

But I'll leave it up to you what to do with it for now. Everything LGTM for you to merge if you'd rather leave it for later

Comment (TJMoats) [reply src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx]: updated to use the buff! Where do you get logs that have shadowlands content? I asked in discord, but I never got an answer :D

Comment (Dambroda) [reply src/parser/priest/holy/modules/talents/75/PrayerCircle.tsx]: Sharqq pointed out that at the very bottom of Warcraftlogs front page you can now see beta logs and search the same way you do with live logs. I didnt find out until today either XD

**Review by Dambroda — APPROVED** [review]
Everything looks good!

**Pull request merged**
